sdk/common/src/main/java/io/opentelemetry/sdk/resources/detectors/ServiceDetector.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.resources.detectors;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.sdk.resources.Entity;
+import io.opentelemetry.sdk.resources.EntityDetector;
+import java.util.Collections;
+import java.util.List;
+
+/** Detects the `service` entity. */
+public final class ServiceDetector implements EntityDetector {
```

Comment:
So two things:

1. namespace + instance.id (and version) are not stable in semantic conventions yet.  This is highly problematic and something we discussed in Entities SIG earlier.  We need to address this, but you need to deal with that (short term) issue now.  This code only leverages the stable semantic conventions.
2. The plan *was* to have (name/namespace) uniquely define a "Service" with version as descriptive.  "ServiceInstance" would be related to Service in that a Service is composed of ServiceInstance.   This is a proposal I have for how to model otel's service within Entities.  I.e. a `service` is a grouping of `service.instance`s.

Reply:
1) Reality is that people read only comment 2 in https://opentelemetry.io/docs/specs/semconv/resource/#service It does not really matter if some people try to restrict it any further. It has been like that since the very beginning and we should accept that it cannot be changed without breaking pretty much everything. While it is somewhat sad that this part is full of missed opportunities, that opportunities are missed unless there is some breaking OpenTelemetry V2.
2) Instances have versions, services do not. If you run a service, you can have multiple instances with different versions in an environment. I do not see what the value of a service entity would be if the service instance entity would provide all the value and would have to copy all the identifying attributes from the service entity.